### PR TITLE
[BACKLOG-37438] Select Folder Dialog - focus on current selected folder

### PR DESCRIPTION
**Pull request by davidmsantos90**
@pentaho/millenniumfalcon @pentaho/wcag please review

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 50s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **398**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-902/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: widgets/src/main/java/org/pentaho/mantle/client/dialogs/folderchooser/FolderTree.java [hunk 1]
```diff
@@ -75,97 +73,35 @@ public class FolderTree extends Tree /*implements IRepositoryFileTreeListener, U
...
 
-    getElement().setId( "solutionTree" ); //$NON-NLS-1$
-    getElement().getStyle().setProperty( "margin", "29px 0px 10px 0px" ); //$NON-NLS-1$ //$NON-NLS-2$
+    DOM.appendChild( element, focusableElement );
+    DOM.sinkEvents( focusableElement, Event.FOCUSEVENTS );
 
-    beforeFetchRepositoryFileTree();
+    this.addSelectionHandler(this::handleItemSelection);
+    this.addOpenHandler(this::handleOpen);
+    this.addCloseHandler(this::handleClose);
```
Comment (dcleao) [inline widgets/src/main/java/org/pentaho/mantle/client/dialogs/folderchooser/FolderTree.java:101]: Formatting is off.

File: widgets/src/main/java/org/pentaho/mantle/client/dialogs/folderchooser/SelectFolderDialog.java [hunk 2]
```diff
@@ -63,18 +63,21 @@ public void onBrowserEvent( Event event ) {
     }
   }
 
-  private static MySolutionTree tree = new MySolutionTree( false );
+  private static final MySolutionTree tree = new MySolutionTree( false );
 
-  public SelectFolderDialog() {
+  private String defaultSelectedPath = FolderTree.getHomeFolder();
+
+  public SelectFolderDialog( String selectedPath ) {
```
Comment (dcleao) [inline widgets/src/main/java/org/pentaho/mantle/client/dialogs/folderchooser/SelectFolderDialog.java:70]: Ideally, we'd add a new constructor and would thus not break compatibility. While you've updated all known Pentaho maintained references, it is never very clear what's public or not, and can be being used by third-parties or not.

**Review by dcleao — CHANGES_REQUESTED** [review]

**Comment by dcleao** [comment]
@davidmsantos90 kudos for the cleanup!

**Review by dcleao — APPROVED** [review]

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 47s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **398**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-902/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by afrjorge — APPROVED** [review]

**Pull request merged**